vm/src/vm.rs

```diff
@@ -944,14 +944,50 @@ impl VirtualMachine {
         }
     }
 
+    // Equivalent to check_class. Masks Attribute errors (into TypeErrors) and lets everything
+    // else go through.
+    fn _check_cls(&self, cls: &PyObjectRef, msg: String) -> PyResult {
+        self.get_attribute(cls.clone(), "__bases__").map_err(|e| {
+            // Only mask AttributeErrors.
+            if e.class().is(&self.ctx.exceptions.attribute_error) {
+                self.new_type_error(msg)
+            } else {
+                e
+            }
+        })
+    }
+
     pub fn _isinstance(&self, obj: &PyObjectRef, cls: &PyObjectRef) -> PyResult<bool> {
```

Comment:
```suggestion
    fn recursive_isinstance(&self, obj: &PyObjectRef, cls: &PyObjectRef) -> PyResult<bool> {
```
Totally looks good to me 😊 How about rename `_isinstance` to `recursive_isinstance`? As cpython and `issubclass` use `recursive_*` naming for that case, I think it is a little more reasonable.

Reply:
Good idea!